### DISCUSSION TITLE
n_offsprings in crossover and algorithm objects

### DISCUSSION THREAD

**Discussion by dinohsu1019** [anyoptimization/pymoo#525] [Q&A]
May I know what's the difference between n_offsprings in a crossover object, say SBX, and one in an algorithm object, say NSGA2? Thanks in advance.

**Comment by blankjul** [comment]
Both times it stands for `Number of Offsprings`. However, the context is different.

In Evolutionary Algorithms (e.g. NSGA2 or DE) it describes how many offsprings should be created in each generation. By default, this is equal to the population size. However, to create less or more greedy versions of algorithms it can be de- or increased.

In the crossover itself, it describes how many individuals will be created during mating. SBX for example requires two parents and will create two offsprings (thus `n_offsprings = 2`). Evolutionary algorithms need to know the meta-information to know how many parents to pass and how many offsprings will be created using them.
